### ISSUE TITLE
Add ``sort_tables`` function

### ISSUE THREAD

**Issue by jeromekelleher** [jeromekelleher/msprime#214]
As discussed in the [comments](https://github.com/jeromekelleher/msprime/pull/213#discussion_r127903251) for PR #123, it would be very useful to have a function that would take a set of tables describing a valid (but non-canonical) tree sequence, and sort the tables such that it is canonical.

Issues:
1. Should we sort the nodes, such that samples are 0...n and we order nodes by time after that? (I think probably not because of complications with ancestral samples and so on, but it's worth thinking about.)
2. We sort sites by increasing position (duplicate sites are not allowed)
3. Do we sort mutation nodes by some topological ordering? I like the idea of pre-order, as it's what naturally happens if you do a Markov generation process down the tree. It would be easy enough to implement here: we just make a tree sequence out of the ``nodes`` and ``edgesets`` tables and then do the required tree traversals for the sites where we have > 1 mutation.

What do you think @petrelharp?

**Comment by petrelharp** [comment]
I think not (1.), because it seems useful to have nodes ordered by time, whether or not they are samples.  Especially if samples were allowed to be internal nodes.

For the remaining questions, here's recollecting what we've said elsewhere:
from #122 for computing statistics (eg freq spectrum)

1. traverse sites along the genome
2. for each mutation find the allele it took the place of

and from #209 for simplify in addition to the above we also need to

3. find all (sites, mutations) that fall on a given node and fall in a given [left, right) interval

So, yes to sorting sites by position, since (1.) is the main thing we'll want to do.

Then, since mutations are associated with sites it is natural to sort them by site.  I don't think the alternative, sorting by node, makes sense.

So then the question is just what order to sort the mutations in for each node.  For this, pre-order seems like a good way to go.  (There was a question of whether to also store the parental mutation, but that doesn't affect the tables.)

Hm, but that argument was for sorting the mutations associated with a site, within a tree sequence.  Here we're just talking about the tables.  A downside to this sorting is that to define the sorting you need the node and edgeset tables - cross-dependency is harder to check.  But, I guess I see how it would make the internal representation much easier to do.

In summary: the second and third requirements sound good, with the caveat that I don't have a good sense of how close the tables are to the internal representation - if the mutation table is processed substantially to get it into the internal representation anyhow, then maybe we don't require the last thing.

It'll be nice to get this sorted and get back to #122.
